Software/Development/SemesterProject/System/scheduler.c

```diff
@@ -16,28 +25,31 @@ taskControlBlock taskList[maxNumberTasks];
 
 void taskListInitialization(void)
 {
-    int taskListIndex = 0;
-    taskList[taskListIndex].functionPointer = task50us;
-    taskList[taskListIndex].taskState = INACTIVE;
-    taskList[taskListIndex].cyclicity = 50;
-    taskListIndex++;
+    Uint16 taskListIndex = 0;
 
     taskList[taskListIndex].functionPointer = task20ms;
     taskList[taskListIndex].taskState = READY;
-    taskList[taskListIndex].cyclicity = 20000;
+    taskList[taskListIndex].cyclicity = 20000;      // Time in us
+    taskList[taskListIndex].timeLeft = 20000;       // Initialize to cyclicity
     taskListIndex++;
 
-    taskList[taskListIndex].functionPointer = NULL;
+    taskList[taskListIndex].functionPointer = task50us;
+    taskList[taskListIndex].taskState = INACTIVE;
+    taskList[taskListIndex].cyclicity = 50;
+    taskList[taskListIndex].timeLeft = 50;
+    taskListIndex++;
+
+    taskList[taskListIndex].functionPointer = NULL; // End of list
 }
 
 void scheduleTasks(void)
 {
-    char taskListIndex = 0;
+    Uint16 taskListIndex = 0;
     for(;;)
```

Comment:
Infinite loop, isn't it possible to integrate this in a function called by the timers?
I know it is not that periodicity is not that important but it seems like this is unnecessary stress on the dsp. Just cycle through the taskList after everything else has been adjusted in the timer's execution.

Reply:
this is the system's infinite loop. There must be an infinite loop somewhere in the system. Otherwise the execution of the code would just finish.

As you see, the infinite loop is constantly checking the task list for tasks which are willing to be executed.